livekit-plugins/livekit-plugins-google/livekit/plugins/google/llm.py

```diff
@@ -0,0 +1,298 @@
+# Copyright 2023 LiveKit, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import inspect
+import json
+import os
+from typing import Any, AsyncIterator, Awaitable, Dict, List, Optional
+from uuid import uuid4 as uuid
+
+import vertexai  # type: ignore
+from livekit.agents import llm
+from livekit.agents.llm._oai_api import (
+    create_ai_function_info,
+)
+from vertexai.generative_models import (  # type: ignore
+    Candidate,
+    FunctionDeclaration,
+    GenerationConfig,
+    GenerationResponse,
+    GenerativeModel,
+    Tool,
+)
+
+from .log import logger
+from .models import GeminiModels
+
+JSON_SCHEMA_TYPE_MAP = {
+    str: "string",
+    int: "integer",
+    float: "number",
+    bool: "boolean",
+    list: "array",
+}
+
+
+class LLM(llm.LLM):
+    def __init__(
+        self,
+        *,
+        project: Optional[str] = None,
+        location: Optional[str] = None,
+        model: str | GeminiModels = "gemini-1.5-flash-002",
+        **kwargs,
+    ) -> None:
+        logger.info(
+            "Initializing LLM with project: %s, location: %s, model: %s",
+            project,
+            location,
+            model,
+        )
+        super().__init__()
+        self._project = project
+        self._location = location
+        self._model_name = model
+        self._kwargs = kwargs
+        self._gac = os.getenv("GOOGLE_APPLICATION_CREDENTIALS")
```

Comment:
Can you follow what we did for auth on the google.STT.

https://github.com/livekit/agents/blob/1a99aae891d6bd440da20f88a3f6e29eab9069cc/livekit-plugins/livekit-plugins-google/livekit/plugins/google/stt.py#L105

As well as the extra arguments for the credentials